osquery/tables/system/linux/pci_devices.cpp

```diff
@@ -26,38 +31,162 @@ const std::string kPCIKeyVendor = "ID_VENDOR_FROM_DATABASE";
 const std::string kPCIKeyModel = "ID_MODEL_FROM_DATABASE";
 const std::string kPCIKeyID = "PCI_ID";
 const std::string kPCIKeyDriver = "DRIVER";
+const std::string kPCISubsysID = "PCI_SUBSYS_ID";
+
+const std::string kPciIdsPath = "/usr/share/misc/pci.ids";
+
+PciDB::PciDB(std::istream& db_filestream) {
+  // pci.ids keep track of subsystem information of vendor and models
+  // sequentially so we keep track of what the current vendor and models are.
+  std::string cur_vendor, cur_model, line;
```

Comment:
It's better variables to have as small scope as possible. cur_vendor and cur_model can be declared in the while loop.

Reply:
Logically, it does not work with `cur_vendor` and `cur_model` in the while loop.  B/c `pci.ids` reads top down, and the models and subsystem always belong to the their respective parents from the lines above, these variables are used to identify the vendor and model the current line belongs to (if vendor or model lines themselves)